django/db/models/fields/__init__.py

```diff
@@ -857,10 +863,42 @@ def _get_default(self):
                 return self.default
             return lambda: self.default
 
+        if self.has_db_default():
+            return lambda: DB_DEFAULT
+
         if not self.empty_strings_allowed or self.null and not connection.features.interprets_empty_strings_as_nulls:
             return return_None
         return str  # return empty string
 
+    def has_db_default(self):
+        """Return a boolean of whether this field has a database default value."""
+        return self.db_default is not NOT_PROVIDED
+
+    def db_default_sql(self, model, schema_editor):
+        from django.db.models.expressions import Expression, Value
+        from django.db.models.functions import DefaultNow
+        from django.db.models.sql.query import Query
+
+        connection = schema_editor.connection
+        if connection.features.supports_functions_in_defaults and isinstance(self.db_default, Expression):
+            db_default = self.db_default
+            default_sql = '(%s)'
+        elif isinstance(self.db_default, DefaultNow):
+            db_default = self.db_default
+            default_sql = '%s'
+        else:
+            db_default = Value(self.db_default)
+            default_sql = '%s'
```

Comment:
The fact you have to special case `DefaultNow` makes it look like a code smell to me, this logic should be completely abstracted by whatever is allowed to be passed to `db_default`.

Reply:
It's a workaround for mysql - in versions before `8.0.13`, it only supports a very limited number of non-literal defaults: https://dev.mysql.com/doc/refman/8.0/en/data-type-defaults.html#data-types-defaults-explicit-old

Reply:
Maybe it should go in `django.contrib.mysql`...

Reply:
That's not really an option as far as I can tell - we can move `DefaultNow`, of course, but the problem here is that we have to special-case how we handle it.